src/System.Private.CoreLib/shared/System/String.Searching.cs

```diff
@@ -105,11 +105,13 @@ public int IndexOfAny(char[] anyOf, int startIndex, int count)
             if (anyOf.Length == 2)
             {
                 // Very common optimization for directory separators (/, \), quotes (", '), brackets, etc
-                return IndexOfAny(anyOf[0], anyOf[1], startIndex, count);
+                var result = SpanHelpers.IndexOfAny(ref Unsafe.Add(ref _firstChar, startIndex), anyOf[0], anyOf[1], count);
```

Comment:
Isn't currently a way to combing the length check for this branch `anyOf.Length == 2` with bounds check elimination for the accesses `anyOf[0], anyOf[1]`?